editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/terrain/AddTerrainViewModel.java

```diff
@@ -17,11 +31,21 @@ public AddTerrainViewModel (Application application) {
 
     // Creates a new instance of terrain and calls addTerrain to add the terrain to the terrain
     // table in the room database
-    public void createNewTerrain(Editable latitudeText, Editable longitudeText) {
+    public void createNewTerrain(Editable latitudeText, Editable longitudeText, Editable heightText, Editable widthText) {
 
         double latitude = Double.parseDouble(latitudeText.toString());
         double longitude = Double.parseDouble(longitudeText.toString());
-        Terrain terrain = new Terrain(latitude, longitude);
+        double height = Double.parseDouble(heightText.toString());
+        double width = Double.parseDouble(widthText.toString());
+
+        Context context = this.getApplication();
+
+        Bitmap heightmap = BitmapFactory.decodeResource(context.getResources(), R.drawable.hardcoded_heightmap);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        heightmap.compress(Bitmap.CompressFormat.PNG, 100, out);
+        byte[] byteHeightmap = out.toByteArray();
+
+        Terrain terrain = new Terrain(latitude, longitude, height, width, byteHeightmap);
```

Comment:
The last argument for Terrain should probably be changed to Drawable or Bitmap

Reply:
Didn't we talk about this on Saturday? We used byte array because we can store it in Room easier?